yazi-core/src/folder/files.rs

```diff
@@ -231,8 +219,15 @@ impl Files {
 			};
 		}
 
-		let (hidden, items) =
-			if self.show_hidden { (vec![], urls) } else { urls.into_iter().partition(|u| u.is_hidden()) };
+		let (hidden, items) = if let Some(filter) = &self.filter {
+			urls.into_iter().partition(|u| {
+				(u.is_hidden() && !self.show_hidden) || !u.file_name().is_some_and(|s| filter.matches(s))
```

Comment:
Major nitpick but if this is getting checked for every file is it a little better to check `self.show_hidden` first every time? This is the same further down too.

```suggestion
				(!self.show_hidden && u.is_hidden()) || !u.file_name().is_some_and(|s| filter.matches(s))
```

Reply:
Nice catch! I forgot to change this check for URLs - it needs to parse the string path, so it's a slow operation. I'll fix it!

But the `f.is_hidden()` for files is intentional - it's a cheap bitwise operation, and because most files aren't hidden, it can short-circuit the `!self.show_hidden`, checking only once.

Reply:
Ah ok cool